whistl/train.py

```diff
@@ -0,0 +1,480 @@
+
+                loss_function = nn.BCEWithLogitsLoss(weight=batch_weights)
+
+                # Standard update step
+                optimizer.zero_grad()
+                output = classifier(expression)
+                loss = loss_function(output, labels)
+                loss.backward()
+                optimizer.step()
+
+                train_loss += loss.item()
+                train_correct += utils.count_correct(output, labels)
+
+            # Disable the gradient and switch into model evaluation mode
+            with torch.no_grad():
+                classifier = classifier.eval()
+
+                tune_loss = 0
+                tune_correct = 0
+                for tune_batch in tune_loader:
+                    expression, labels, ids = tune_batch
+                    expression = expression.float().to(device)
+                    tune_labels = labels.float().to(device)
+
+                    batch_weights = [class_weights[int(label)] for label in labels]
+                    batch_weights = torch.FloatTensor(batch_weights).to(device)
+
+                    loss_function = nn.BCEWithLogitsLoss(weight=batch_weights)
+
+                    tune_output = classifier(expression)
+                    loss = loss_function(tune_output, tune_labels)
+                    tune_loss += loss.item()
+                    tune_correct += utils.count_correct(tune_output, tune_labels)
+
+                # Save the model
+                if save_file is not None:
+                    if best_tune_loss is None or tune_loss < best_tune_loss:
+                        best_tune_loss = tune_loss
+                        torch.save(classifier, save_file)
+
+            train_accuracy = train_correct / len(train_dataset)
+            tune_accuracy = tune_correct / len(tune_dataset)
+
+            if logger is not None:
+                logger.info('Epoch {}'.format(epoch))
+                logger.info('Train loss: {}'.format(train_loss / len(train_dataset)))
+                logger.info('Tune loss: {}'.format(tune_loss / len(tune_dataset)))
+                logger.info('Train accuracy: {}'.format(train_accuracy))
+                logger.info('Tune accuracy: {}'.format(tune_accuracy))
+                logger.info('Baseline accuracy: {}'.format(baseline))
+
+            results['train_loss'].append(train_loss / len(train_dataset))
+            results['tune_loss'].append(tune_loss / len(tune_dataset))
+            results['train_acc'].append(train_accuracy)
+            results['tune_acc'].append(tune_accuracy)
+    except Exception as e:
+        # Print error
+        logger.error(e, exc_info=True)
+    finally:
+        results = utils.add_study_ids_to_results(results, train_dirs, tune_dirs)
+        return results
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='This script trains a classifier to differentiate'
+                                                 ' between sepsis and healthy gene expresssion')
```

Comment:
You might want to update this description at some point (if I remember correctly you're trying to generalize this beyond just sepsis to other diseases like TB and lupus, right?)

Reply:
Good catch, I went ahead and updated the file header comment too to be more up to date